src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,715 @@
+            if format_ == 'json':
+                config_data = json.load(config_file)
+                if 'FeatureManagement' in config_data:
+                    features_dict = config_data['FeatureManagement']
+            elif format_ == 'yaml':
+                logger.warning("Importing feature flags from a yaml file is not supported yet. Ignoring all feature flags.")
+            elif format_ == 'properties':
+                logger.warning("Importing feature flags from a properties file is not supported yet. Ignoring all feature flags.")
+
+    except ValueError:
+        raise CLIError(
+            'The input is not a well formatted %s file.' % (format_))
+    except OSError:
+        raise CLIError('File is not available.')
+
+    # features_dict contains all features that need to be converted to KeyValue format now
+    return __convert_feature_dict_to_keyvalue_list(features_dict, format_)
+
+
+def __write_kv_and_features_to_file(file_path, key_values=None, features=None, format_=None, separator=None, skip_features=False):
+    try:
+        exported_keyvalues = __export_keyvalues(key_values, format_, separator, None)
+        if features and not skip_features:
+            exported_features = __export_features(features, format_)
+            exported_keyvalues.update(exported_features)
+
+        with open(file_path, 'w') as fp:
+            if format_ == 'json':
+                json.dump(exported_keyvalues, fp, indent=2, ensure_ascii=False)
+            elif format_ == 'yaml':
+                yaml.dump(exported_keyvalues, fp, sort_keys=False)
+            elif format_ == 'properties':
+                javaproperties.dump(exported_keyvalues, fp)
+    except Exception as exception:
+        raise CLIError("Failed to export key-values to file. " + str(exception))
+
+
+# Config Store <-> List of KeyValue object
+
+
+def __read_kv_from_config_store(cmd, name=None, connection_string=None, key=None, label=None, prefix_to_remove="", prefix_to_add=""):
+    from .keyvalue import list_key
+    try:
+        # fetch complete keyvalue list
+        key_values = list_key(cmd,
+                              key=key,
+                              label=QueryKeyValueCollectionOptions.empty_label if label is None else label,
+                              name=name,
+                              connection_string=connection_string,
+                              all_=True)
+        # add prefix, remove label, remove non-user-info attributes
+        for kv in key_values:
+            # remove prefix if specified
+            if not kv.key.startswith(prefix_to_remove):
+                continue
+            kv.key = kv.key[len(prefix_to_remove):]
+            # add prefix if specified
+            kv.key = prefix_to_add + kv.key
+    except Exception as exception:
+        raise CLIError(str(exception))
+    return key_values
+
+
+def __read_features_from_config_store(cmd, name=None, connection_string=None, key=None, label=None):
```

Comment:
>__read_features_from_config_store [](start = 4, length = 33)

this helper method looks unnecessary. caller can call list_feature directly. Also it is odd to let this internal helper method to depend on feature.py.

Reply:
done

---
In reply to: [356883121](https://github.com/Azure/azure-cli/pull/11467#discussion_r356883121) [](ancestors = 356883121)